integrations/google_genai/src/haystack_integrations/components/generators/google_genai/chat/chat_generator.py

```diff
@@ -469,24 +637,105 @@ def _convert_google_chunk_to_streaming_chunk(
+        """
+        Aggregate streaming chunks into a final ChatMessage with reasoning content and thought signatures.
+
+        This method extends the standard streaming chunk aggregation to handle Google GenAI's
+        specific reasoning content, thinking token usage, and thought signatures for multi-turn context.
+
+        :param chunks: List of streaming chunks to aggregate.
+        :returns: Final ChatMessage with aggregated content, reasoning, and thought signatures.
+        """
+
+        # Use the generic aggregator for standard content (text, tool calls, basic meta)
+        message = _convert_streaming_chunks_to_chat_message(chunks)
+
+        # Now enhance with Google-specific features: reasoning content, thinking token usage, and thought signatures
+        reasoning_text_parts: list[str] = []
+        thought_signatures: List[Dict[str, Any]] = []
+        thoughts_token_count = None
+
+        for chunk in chunks:
+            # Extract reasoning deltas
+            if chunk.meta and "reasoning_deltas" in chunk.meta:
+                reasoning_deltas = chunk.meta["reasoning_deltas"]
+                if isinstance(reasoning_deltas, list):
+                    for delta in reasoning_deltas:
+                        if delta.get("type") == "reasoning":
+                            reasoning_text_parts.append(delta.get("content", ""))
+
+            # Extract thought signature deltas (for multi-turn context preservation)
+            if chunk.meta and "thought_signature_deltas" in chunk.meta:
+                signature_deltas = chunk.meta["thought_signature_deltas"]
+                if isinstance(signature_deltas, list):
+                    # Aggregate thought signatures - they should come from the final chunks
+                    # We'll keep the last set of signatures as they represent the complete state
+                    thought_signatures = signature_deltas
+
+            # Extract thinking token usage (from the last chunk that has it)
+            if chunk.meta and "usage" in chunk.meta:
+                chunk_usage = chunk.meta["usage"]
+                if "thoughts_token_count" in chunk_usage:
+                    thoughts_token_count = chunk_usage["thoughts_token_count"]
+
+        # Add thinking token count to usage if present
+        if thoughts_token_count is not None and "usage" in message.meta:
+            if message.meta["usage"] is None:
+                message.meta["usage"] = {}
+            message.meta["usage"]["thoughts_token_count"] = thoughts_token_count
+
+        # Add thought signatures to meta if present (for multi-turn context preservation)
+        if thought_signatures:
+            message.meta["thought_signatures"] = thought_signatures
+
+        # If we have reasoning content, reconstruct the message to include it
+        # Note: ChatMessage doesn't support adding reasoning after creation, reconstruction is necessary
```

Comment:
(just responding to the comment: you can access and manipulate the `_content` attribute of `ChatMessage` if you prefer, but I am OK with your implementation)